### PR TITLE
Remove image preflight nodeinit.

### PR DESCRIPTION
**Pull request by axel7born**
**How to categorize this PR?**

/area networking
/kind cleanup

**What this PR does / why we need it**:
Cleanup unneeded container images node-init and preflight.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Cleanup unneeded container images etcd-operator, node-init and preflight.
```

### CODE REVIEW

File: charts/internal/cilium/requirements.yaml [hunk 1]
```diff
@@ -5,13 +5,9 @@ dependencies:
...
-  - name: preflight
-    condition: requirements.preflight.enabled
-  - name: nodeinit
-    condition: global.nodeinit.enabled
   - name: managed-etcd
     condition: global.etcd.managed
   - name: hubble-ui
     condition: requirements.hubble.enabled
   - name: hubble-relay
-    condition: requirements.hubble.enabled
```
Comment (ScheererJ) [inline charts/internal/cilium/requirements.yaml:17]: Nit: Changing the end-of-line was not really necessary, but does not hurt either.

File: pkg/cilium/types.go [hunk 2]
```diff
@@ -26,11 +26,6 @@ const (
 	CiliumOperatorImageName = "cilium-operator"
 	// CiliumETCDOperatorImageName defines cilium etcd operator image name.
 	CiliumETCDOperatorImageName = "cilium-etcd-operator"
-	// CiliumNodeInitImageName defines cilium's node init image name.
-	CiliumNodeInitImageName = "cilium-nodeinit"
-	// CiliumPreflightImageName defines the cilium preflight image name.
-	CiliumPreflightImageName = "cilium-preflight"
-
```
Comment (ScheererJ) [inline pkg/cilium/types.go:33]: Can you please leave the empty line as delimiter between the cilium and hubble images?

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Looks mostly fine.
What about the `etcd-operator` container? Should we remove it in a separate pull request or include it here?

File: charts/internal/cilium/values.yaml [hunk 3]
```diff
@@ -490,9 +475,7 @@ global:
   images:
     cilium-agent: "image-repository:image-tag"
     cilium-etcd-operator: "image-repository:image-tag"
```
Comment (ScheererJ) [inline charts/internal/cilium/values.yaml:477]: Please remove this line as well.

```suggestion
```

File: pkg/charts/utils.go [hunk 4]
```diff
@@ -193,9 +190,8 @@ func generateChartValues(config *ciliumv1alpha1.NetworkConfig, network *extensio
 
 	// If ETCD enabled
 	if config.Store != nil {
-		if *config.Store == ciliumv1alpha1.ETCD {
-			globalConfig.Etcd.Enabled = true
-			globalConfig.Etcd.Managed = true
+		if *config.Store == ciliumv1alpha1.Kubernetes {
+			return requirementsConfig, globalConfig, fmt.Errorf("%s is not a supported value for field store", *config.Store)
```
Comment (ScheererJ) [inline pkg/charts/utils.go:194]: Should this really be an equal check? I think we always use kubernetes as backing store. Hence, anything not equal to kubernetes would be not supported...

**Review by ScheererJ — CHANGES_REQUESTED** [review]

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Please also remove the `etcd` section in `values.yaml`.

File: charts/internal/cilium/values.yaml (hunk 3 above)
Comment (axel7born) [reply charts/internal/cilium/values.yaml]: Removed.

File: pkg/charts/utils.go (hunk 4 above)
Comment (axel7born) [reply pkg/charts/utils.go]: Reversed the condition.

**Comment by axel7born** [comment]
> Please also remove the `etcd` section in `values.yaml`.

Done.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**